### PR TITLE
Maintenance page

### PR DESCRIPTION
**Pull request by struds**
## Context

Terraform and Workflow to build and deploy a maintenance page to any environment

## Changes in this pull request

- Added maintenance-page directory with it's own terraform backend and workspaces - this is decoupled from any other deployment of the app and infrastructure.
- Added static index page
- Added a manual workflow to trigger maintenance page deployment

## Testing

Follow the maintenance-page/README instructions 

The CloudFront URL for staging:

 https://dxg9e7r3hongn.cloudfront.net

Test the URL with a path shows the service down page and not nasty XML.

https://dxg9e7r3hongn.cloudfront.net/dfgfdgfdg
https://dxg9e7r3hongn.cloudfront.net/account/your-beacon-registry-account

You cannot seem to test the workflow until it is merged to 'main' due to a long standing GitHub feature/issue :-) Will have to test and tweak if needed after merge :-/

### CODE REVIEW

**Comment by struds** [comment]
musing to myself.. I wonder if the error page should always redirect to index.html i.e the service is down page. otherwise it;s confusing to browse to:

https://<real url>/account/your-beacon-registry-account

and see 404 instead service down

When browsing to:

https://<real url>/sdfgdfsgfg

it makes sense to see the error page but also makes sense to see the down page.. so maybe there is never an error page only the service down page whatever the user does.

File: maintenance-page/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+The terraform files within this directory represents the infrastructure-as-code for the Beacons Maintenance Page.
+
+The basic premise is a static https enabled page served via CloudFront and S3 that can be swapped via the route53 alias functionality.
+
+You can build a maintenance page for any environment or URL although the default settings tie together the environment and the urls currently in use.
+
+e.g the staging environment uses the url staging.406beacons.com.
+
+Note: the s3 bucket 'must' have be named to match the hostname used to access the page e.g staging.406beacons.com
```
Comment (Ninamma) [inline maintenance-page/README.md:11]:
```suggestion
Note: the s3 bucket 'must' be named to match the hostname used to access the page e.g staging.406beacons.com
```

File: maintenance-page/production.tfvars [hunk 2]
```diff
@@ -0,0 +1,2 @@
+aws_region             = "eu-west-2"
+cloudfront_domain_name = "backoffice.production.406beacons.com"
```
Comment (Ninamma) [inline maintenance-page/production.tfvars:2]:
```suggestion
cloudfront_domain_name = "production.406beacons.com"
```
not sure if this is on purpose to not mess with production while testing this?

**Review by Ninamma — APPROVED** [review]
Looks okay to me - apart from a couple teeny things I commented on.

Once the Github Actions is tested and sorted, maybe more information in the README would be good to clarify how to deploy the page in the Pipeline vs how to do it manually as described in the README right now?

File: maintenance-page/production.tfvars (hunk 2 above)
Comment (struds) [reply maintenance-page/production.tfvars:2]: Yep - thats a mistake.. copied over the back office setup and this didn't get changed.

**Pull request merged**